src/arizona_template_parser.erl

```diff
@@ -0,0 +1,287 @@
+-module(arizona_template_parser).
+-moduledoc false.
+
+%% --------------------------------------------------------------------
+%% API function exports
+%% --------------------------------------------------------------------
+
+-export([parse/1]).
+
+%
+
+-ignore_xref([parse/1]).
+
+%% --------------------------------------------------------------------
+%% Types (and their exports)
+%% --------------------------------------------------------------------
+
+-opaque element() :: {text, arizona_template_scanner:location(), binary()}
+                   | {expr, arizona_template_scanner:location(), binary()}
+                   | {block, arizona_template_scanner:location(), block()}
+                   | {tag, arizona_template_scanner:location(), tag()}.
+-export_type([element/0]).
+
+-type block() :: #{
```

Comment:
Is this purposefully not opaque?

Reply:
If it's internal this either goes to show we should move some code around or opacity-breakage is Ok. Since we don't yet have a defined interface, bear with me while I try to understand what should/shouldn't be opaque. (i.e. don't do more than is required, now, since we'll have time to make more stuff opaque if need be).